jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/common/Badge.java

```diff
@@ -1,45 +1,99 @@
 package com.epam.jdi.light.angular.elements.common;
 
 import com.epam.jdi.light.angular.asserts.BadgeAssert;
+import com.epam.jdi.light.angular.elements.enums.BadgePosition;
+import com.epam.jdi.light.angular.elements.enums.BadgeSize;
 import com.epam.jdi.light.common.JDIAction;
 import com.epam.jdi.light.elements.base.UIBaseElement;
 import com.epam.jdi.light.elements.common.UIElement;
+import com.epam.jdi.light.elements.interfaces.base.HasClick;
+import com.epam.jdi.light.elements.interfaces.common.IsText;
+
+import static com.epam.jdi.light.angular.elements.enums.BadgeSize.LARGE;
+import static com.epam.jdi.light.angular.elements.enums.BadgeSize.MEDIUM;
+import static com.epam.jdi.light.angular.elements.enums.BadgeSize.SMALL;
 
 /**
  * To see an example of Badge web element please visit https://material.angular.io/components/badge/overview.
  */
 
-public class Badge extends UIBaseElement<BadgeAssert> {
+public class Badge extends UIBaseElement<BadgeAssert> implements HasClick, IsText {
 
     @JDIAction("Get '{name}' badge")
-    public UIElement badge() {
-        return this.find(".mat-badge-content");
-    }
-
-    @JDIAction("'{name}' has color '{0}'")
-    public boolean color(String color) {
-        return getColor().equalsIgnoreCase(color);
+    private UIElement badge() {
+        return core().find(".mat-badge-content");
```

Comment:
If the locator is named content, it is better to use the same name for variable. 
Absolutely not clear why do we have badge inside the badge.

Reply:
Angular have absolute different approach to badge, in Angular it should be an inferface!